### ISSUE TITLE
CJK improvement by listening keyboard input events

### ISSUE THREAD

**Issue by yo-goto** [tgrosinger/advanced-tables-obsidian#53]
Hello, @tgrosinger. I'm from here https://github.com/tgrosinger/advanced-tables-obsidian/issues/16
In Obsidian v0.10.2, the feature of keyboard input event listener to avoid the Enter key collision with IME was added that I actually had requested to Licat and Silver. If the API supports this, please consider adding the feature to enable to move to a next cell by Enter key with IME as I mentioned in the link above.

Thank you.

**Comment by tgrosinger** [comment]
Hi @yo-goto, I read through the forum post you linked, and I understand the problem but I am not clear on the solution. I actually created #47 to see if there was something I could do here, but I haven't been able to find out the answer yet.

In the Obsidian API, the Hotkeys are defined with no information about what they should contain.

```
/**
 * @public
 */
export interface Hotkey {

}
```

To add to the complexity, I am not actually using Obsidian hotkeys for binding <kbd>enter</kbd> and <kbd>tab</kbd> in a table. Obsidian does not have a way of triggering a plugin on non-modified hotkeys (not using ctrl, alt, or meta). To get around that I am binding directly on CodeMirror.

So long story short, I'd want to make this a better experience, but I'm not sure how to go about that yet.

**Comment by yo-goto** [comment]
Thanks for the replay. I didn't expect using hotkeys. I assumed that KeymapContext, KeyEventHandler and KeymapEventListener implemented in 0.10.2 may be somehow related with the solution.  Can plugins listen the keymap events by using those types? But, sorry, I don't know about the detail. 

https://github.com/obsidianmd/obsidian-api/commit/d10f2f6efc0d0d7c9bf96cd435ef376b18fbd6d8#diff-0eaea5db2513fdc5fe65d534d3591db5b577fe376925187c8a624124632b7466

I'm going to ask about this for someone familiar with Japanese key input implementation.

**Comment by yo-goto** [comment]
This may be helpful. Inkdrop plugin and event-kit
https://github.com/jmerle/inkdrop-table-editor
https://github.com/atom/event-kit
